### PR TITLE
Task s4 avatar movement

### PR DESCRIPTION
**Pull request by karimkefi**
Task avatar movement into story4!

### CODE REVIEW

File: includes/avatarMove.html [hunk 1]
```diff
@@ -0,0 +1,16 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
+    <title>Avatar Move</title>
+</head>
+<body>
+
+<svg id="svgOne" xmlns="http://www.w3.org/2000/svg" width="800" height="600" class="svgbox">
```
Comment (ailindoyle) [inline includes/avatarMove.html]: does this need indenting?
does line spacing need to be tidied up?

File: js/avatarMovement.js [hunk 2]
```diff
@@ -0,0 +1,148 @@
+
+//player A variables
+var playerAx = 150;
+var playerAy = 150;
+var playerAHeight = 50;
+var playerAWidth = 50;
+var playerAOrientation='E';
```
Comment (ailindoyle) [inline js/avatarMovement.js]: space before =

File: js/avatarMovement.js [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+
+//Player B variables
+var playerBx = 550;
+var playerBy = 550;
+var playerBHeight = 50;
+var playerBWidth = 50;
+var playerBOrientation='W';
```
Comment (ailindoyle) [inline js/avatarMovement.js]: space before =

File: js/avatarMovement.js [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+    rect.setAttributeNS(null, 'orientation',orientation)
+    document.getElementById('svgOne').appendChild(rect);
+    return document.getElementById(id);
+}
+
+
+//  <-  37 = left arrow,  39 = right arrow , 38 = up arrow, 40 = down arrow
+//  <-  65 = A (left) ,  68 = D (right)  , 87 = W (up) , 83 = S (down)
+function keyDownHandler(e) {
+    $playerA=$(playerA);
```
Comment (ailindoyle) [inline js/avatarMovement.js]: spaces around =

File: js/avatarMovement.js [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+    $playerB=$(playerB);
```
Comment (ailindoyle) [inline js/avatarMovement.js]: spaces around =

File: js/avatarMovement.js [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+            moveLeft($playerA);
+            orientation(playerA,'W');
+            break;
+        case 39:
+            moveRight($playerA);
+            orientation(playerA,'E');
+            break;
+        case 38:
+            moveUp($playerA);
+            playerAOrientation='N';
```
Comment (ailindoyle) [inline js/avatarMovement.js]: space before =
for all playerXOrientation lines...

File: js/avatarMovement.js [hunk 7]
```diff
@@ -0,0 +1,148 @@
...
+}
+
+
+//  <-  37 = left arrow,  39 = right arrow , 38 = up arrow, 40 = down arrow
+//  <-  65 = A (left) ,  68 = D (right)  , 87 = W (up) , 83 = S (down)
+function keyDownHandler(e) {
+    $playerA=$(playerA);
+    $playerB=$(playerB);
+    switch(e.keyCode) {
+        case 37:
```
Comment (ailindoyle) [inline js/avatarMovement.js]: does this need playerAOrientation = 'W'?

File: js/avatarMovement.js [hunk 8]
```diff
@@ -0,0 +1,148 @@
...
+            moveLeft($playerA);
+            orientation(playerA,'W');
+            break;
+        case 39:
```
Comment (ailindoyle) [inline js/avatarMovement.js:46]: does this need playerAOrientation = 'E'?

File: js/avatarMovement.js [hunk 9]
```diff
@@ -0,0 +1,148 @@
...
+            moveUp($playerB);
+            playerBOrientation='N';
+            orientation(playerB,'N');
+            break;
+        case 83:
+            moveDown($playerB);
+            playerBOrientation='S';
+            orientation(playerB,'S');
+            break;
+        default:
```
Comment (ailindoyle) [inline js/avatarMovement.js]: what is this default?

File: js/avatarMovement.js [hunk 10]
```diff
@@ -0,0 +1,148 @@
...
+
+
+function moveLeft($player) {
+    if (!$player.hasClass('moving')) {
+        $player.addClass('moving');
+        $player.animate({x: "-="+distance}, time ,'linear');
+    }
+}
+
+function orientation(player, direction) {
```
Comment (ailindoyle) [inline js/avatarMovement.js]: move function above or below all the moveDirection functions in the code

File: js/avatarMovement.js [hunk 11]
```diff
@@ -0,0 +1,148 @@
...
+        player.setAttributeNS(null, 'orientation',direction)
```
Comment (ailindoyle) [inline js/avatarMovement.js]: needs space before 'direction'

File: js/avatarMovement.js [hunk 12]
```diff
@@ -0,0 +1,148 @@
...
+}
+
+function moveDown($player) {
+    if (!$player.hasClass('moving')) {
+        $player.addClass('moving');
+        $player.animate({y: "+="+distance}, time ,'linear');
+    }
+}
+
+function stopMovement (player){
```
Comment (ailindoyle) [inline js/avatarMovement.js]: space between (player) and { required

File: js/avatarMovement.js [hunk 13]
```diff
@@ -0,0 +1,148 @@
...
+    $(player).removeClass('moving');
+    $(player).stop()
+}
+
+
+//event listener for player's avatar
+document.addEventListener("keydown", keyDownHandler, false);
```
Comment (ailindoyle) [inline js/avatarMovement.js]: what does the false do here?

File: js/avatarMovement.js [hunk 14]
```diff
@@ -0,0 +1,148 @@
+
```
Comment (ailindoyle) [inline js/avatarMovement.js]: line breaks between sections of code are uneven - make sure these are consistent

**Review by ailindoyle — CHANGES_REQUESTED** [review]

File: includes/avatarMove.html (hunk 1 above)
Comment (karimkefi) [reply includes/avatarMove.html]: done

File: js/avatarMovement.js (hunk 3 above)
Comment (karimkefi) [reply js/avatarMovement.js]: done

File: js/avatarMovement.js (hunk 4 above)
Comment (karimkefi) [reply js/avatarMovement.js]: done

File: js/avatarMovement.js (hunk 5 above)
Comment (karimkefi) [reply js/avatarMovement.js]: done

File: js/avatarMovement.js (hunk 7 above)
Comment (karimkefi) [reply js/avatarMovement.js]: done

File: js/avatarMovement.js (hunk 6 above)
Comment (karimkefi) [reply js/avatarMovement.js]: done

File: js/avatarMovement.js (hunk 7 above)
Comment (karimkefi) [reply js/avatarMovement.js]: updated

File: js/avatarMovement.js (hunk 8 above)
Comment (karimkefi) [reply js/avatarMovement.js:46]: updated

File: js/avatarMovement.js (hunk 9 above)
Comment (karimkefi) [reply js/avatarMovement.js]: removed

File: js/avatarMovement.js (hunk 10 above)
Comment (karimkefi) [reply js/avatarMovement.js]: done

File: js/avatarMovement.js (hunk 11 above)
Comment (karimkefi) [reply js/avatarMovement.js]: done

File: js/avatarMovement.js (hunk 12 above)
Comment (karimkefi) [reply js/avatarMovement.js]: done

File: js/avatarMovement.js (hunk 13 above)
Comment (karimkefi) [reply js/avatarMovement.js]: removed

File: js/avatarMovement.js (hunk 14 above)
Comment (karimkefi) [reply js/avatarMovement.js]: done

**Review by ailindoyle — APPROVED** [review]
Looks good

**Pull request merged**